src/v/storage/segment.cc

```diff
@@ -569,24 +569,22 @@ auto with_segment(ss::lw_shared_ptr<segment> s, Func&& f) {
 }
 
 ss::future<ss::lw_shared_ptr<segment>> open_segment(
-  std::filesystem::path path,
+  segment_full_path path,
   debug_sanitize_files sanitize_fileops,
   std::optional<batch_cache_index> batch_cache,
   size_t buf_size,
   unsigned read_ahead,
   storage_resources& resources) {
-    auto const meta = segment_path::parse_segment_filename(
-      path.filename().string());
-    if (!meta || meta->version != record_version_type::v1) {
+    if (path.get_version() != record_version_type::v1) {
```

Comment:
the simplification here is that we arrive with a valid full path so no need to replicate the `!meta` check?

Reply:
right, no re-parse, no re-check